crates/red_knot_python_semantic/src/types/unpacker.rs

```diff
@@ -142,29 +168,40 @@ impl<'db> Unpacker<'db> {
     /// Returns the [`Type`] elements inside the given [`TupleType`] taking into account that there
     /// can be a starred expression in the `elements`.
     fn tuple_ty_elements(
-        &mut self,
+        &self,
+        expr: &ast::Expr,
         targets: &[ast::Expr],
```

Comment:
Additional changes:
* Expanded the comments in this method
* Avoid `&mut self` as it's not required (not sure why clippy didn't catch that)
* Pass the `expr` which is required for the diagnostic location